src/overlays/actors/ovl_En_Grasshopper/z_en_grasshopper.h

```diff
@@ -2,16 +2,78 @@
 #define Z_EN_GRASSHOPPER_H
 
 #include "global.h"
+#include "objects/object_grasshopper/object_grasshopper.h"
+
+#define EN_GRASSHOPPER_GET_TYPE(thisx) ((thisx)->params)
 
 struct EnGrasshopper;
 
 typedef void (*EnGrasshopperActionFunc)(struct EnGrasshopper*, PlayState*);
 
+typedef enum EnGrasshopperType {
+    /* -1 */ EN_GRASSHOPPER_TYPE_UNUSED_NORMAL = -1, // Acts exactly like EN_GRASSHOPPER_TYPE_NORMAL
+    /*  0 */ EN_GRASSHOPPER_TYPE_NORMAL,
+    /*  1 */ EN_GRASSHOPPER_TYPE_GROWS_WHEN_SPAWNED, // Spawned by EnEncount1
+    /*  2 */ EN_GRASSHOPPER_TYPE_WOODFALL_TEMPLE_FINAL_ROOM,
+    /*  3 */ EN_GRASSHOPPER_TYPE_WOODFALL,
+} EnGrasshopperType;
+
+typedef struct EnGrasshopperEffect {
+    /* 0x00 */ u8 isEnabled;
+    /* 0x04 */ Vec3f pos;
+    /* 0x10 */ Vec3f velocity;
+    /* 0x1C */ f32 yaw;
+    /* 0x20 */ Vec3f scale;
+    /* 0x2C */ s16 timer;
+    /* 0x2E */ s16 lightningIndex;
+} EnGrasshopperEffect; // size = 0x30
+
+#define EN_GRASSHOPPER_EFFECT_COUNT 100
+
 typedef struct EnGrasshopper {
     /* 0x0000 */ Actor actor;
-    /* 0x0144 */ char unk_144[0x164];
+    /* 0x0144 */ SkelAnime skelAnime;
+    /* 0x0188 */ Vec3s jointTable[DRAGONFLY_LIMB_MAX];
+    /* 0x0218 */ Vec3s morphTable[DRAGONFLY_LIMB_MAX];
     /* 0x02A8 */ EnGrasshopperActionFunc actionFunc;
-    /* 0x02AC */ char unk_2AC[0x15C4];
+    /* 0x02AC */ u8 decision;
+    /* 0x02AD */ u8 shouldTurn;
+    /* 0x02AE */ char unk_2AE[2];
```

Comment:
```suggestion
    /* 0x02AE */ UNK_TYPE1 unk_2AE[2];
```
Same for the other 2 please :)